posthog/health.py

```diff
@@ -64,30 +89,42 @@ def readyz(request):
     web server does a lot of stuff, and kafka is only used I believe for sending
     merge person events, so we'd rather stay up with degraded functionality,
     rather than take the website UI down.
+
+    We also accept an optional `role` parameter which can be any `ServiceRole`,
+    and can be used to specify that a subset of dependencies should be checked,
+    specific to the role a process is playing.
     """
     exclude = set(request.GET.getlist("exclude", []))
     role = request.GET.get("role", None)
 
-    if role and role not in get_args(ServiceRole):
-        return JsonResponse({"error": "InvalidRole"}, status=400)
-
-    checks = {
-        "http": True,
-        "postgres": is_postgres_connected(),
-        "postgres_migrations_uptodate": are_postgres_migrations_uptodate(),
-        "kafka_connected": is_kafka_connected(),
+    available_checks = {
+        "clickhouse": is_clickhouse_connected,
+        "postgres": is_postgres_connected,
+        "postgres_migrations_uptodate": are_postgres_migrations_uptodate,
+        "kafka_connected": is_kafka_connected,
+        "celery_broker": is_celery_broker_connected,
+        "cache": is_cache_backend_connected,
     }
 
     if role:
+        if role not in get_args(ServiceRole):
```

Comment:
I find it more readable like it was before (despite you've now optimized an `if`)